### PR TITLE
109 - Add totalFilesCount to GetDatasetFiles use case

### LINKED ISSUE #109: Add total file count to GetDatasetFiles use case
**Overview of the Feature Request**

As for the Dataset previews, the GetDatasetFiles should be extended to return the total file count, which is necessary for pagination.

We should follow the same implementation as for DatasetPreviewSubset.

This feature depends on: https://github.com/IQSS/dataverse/issues/10202

**What kind of user is the feature intended for?**

- Use case consumer

**What inspired the request?**

- https://github.com/IQSS/dataverse-client-javascript/pull/107#discussion_r1434858529

**What existing behavior do you want changed?**

GetDatasetFiles use case to include total file count

**Any brand new behavior do you want to add to Dataverse?**

None

**Any open or closed issues related to this feature request?**

- https://github.com/IQSS/dataverse/issues/10202

### PR DESCRIPTION
**Pull request by MellyGray**
## What this PR does / why we need it:
This PR introduces `totalFilesCount` to the `GetDatasetFiles` use case, which is essential for pagination. This ensures consistency with the pagination implementation, as seen in `DatasetPreviewSubset`.

## Which issue(s) this PR closes:

- Closes #109 

## Related Dataverse PRs:

- Depends on https://github.com/IQSS/dataverse/issues/10202

## Special notes for your reviewer:
The implementation should be consistent with GetAllDatasetPreviews implementation, so just compare the code with that use case

## Suggestions on how to test this:
Run integration tests or review GitHub action execution

### CODE REVIEW

**Review by GPortas — CHANGES_REQUESTED** [review]
Looks good.

We need also to export the `FilesSubset` model in the `index.ts` file of the files folder, [just as we do with the `DatasetPreviewSubset`](https://github.com/IQSS/dataverse-client-javascript/blob/develop/src/datasets/index.ts#L47)

File: src/files/infra/repositories/transformers/fileTransformers.ts [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
-  const files: File[] = [];
+export const transformFilesResponseToFilesSubset = (
+    response: AxiosResponse,
+): FilesSubset => {
   const filesPayload = response.data.data;
+  const files: File[] = [];
   // eslint-disable-next-line  @typescript-eslint/no-explicit-any
-  filesPayload.forEach(function (filePayload: any) {
-    files.push(transformFilePayloadToFile(filePayload));
+  filesPayload.forEach(function (filesPayload: any) {
```
Comment (GPortas) [inline src/files/infra/repositories/transformers/fileTransformers.ts]: Why have you changed the name of the variable inside the function from singular to plural?

Now we have `transformFilePayloadToFile(filesPayload)` which makes it a little confusing.

**Comment by MellyGray** [comment]
> Looks good.
> 
> We need also to export the `FilesSubset` model in the `index.ts` file of the files folder, [just as we do with the `DatasetPreviewSubset`](https://github.com/IQSS/dataverse-client-javascript/blob/develop/src/datasets/index.ts#L47)

I addressed the change. But I'm worried about what would had happened if you hadn't noticed. I think that somehow we need some tests to test the package from the outside. To force the check on these index exports, and  also ensure that the package functions properly when used as a library

**Comment by GPortas** [comment]
> > Looks good.
> > We need also to export the `FilesSubset` model in the `index.ts` file of the files folder, [just as we do with the `DatasetPreviewSubset`](https://github.com/IQSS/dataverse-client-javascript/blob/develop/src/datasets/index.ts#L47)
> 
> I addressed the change. But I'm worried about what would had happened if you hadn't noticed. I think that somehow we need some tests to test the package from the outside. To force the check on these index exports, and also ensure that the package functions properly when used as a library

Yes. There is an open issue related to functional tests that would be useful for what you mention: https://github.com/IQSS/dataverse-client-javascript/issues/81

I just updated the issue to link to this comment.

**Review by GPortas — COMMENTED** [review]
Sorry, I forgot to submit this review.

Can you check the comment?

File: src/files/infra/repositories/transformers/fileTransformers.ts (hunk 1 above)
Comment (MellyGray) [reply src/files/infra/repositories/transformers/fileTransformers.ts]: My bad, I committed the change to restore the name

**Review by GPortas — APPROVED** [review]

**Pull request merged**